### PR TITLE
Windows: Add dark theme support.

### PR DESCRIPTION
**Pull request by moko256**
This PR adds the dark theme support for both `windows` and `winuwp` engine, based on flutter/flutter#54612.

On Win32: using a registry value that is not documented. (I tested on Windows 10.0.19043.1165)
On UWP: using default background color. (I couldn't find WinRT API that returns dark/light directly, without xaml package.)

TODO:
- [X] Provide the theme value
- [X] Notify when the theme value is changed.

Flutter PR for win32: flutter/flutter#88520.
This PR will close flutter/flutter#54612.
This PR will complete one item in flutter/flutter#70214.

No changes in `flutter/tests`.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

### CODE REVIEW

File: shell/platform/windows/system_utils_win32.cc [hunk 1]
```diff
@@ -106,4 +106,26 @@ bool Prefer24HourTime(std::wstring time_format) {
...
+      L"Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize",
+      L"AppsUseLightTheme", RRF_RT_REG_DWORD, nullptr, &use_light_theme,
+      &use_light_theme_size);
+
+  if (result == 0) {
+    if (use_light_theme == 1) {
+      return kPlatformBrightnessLight;
+    } else {
+      return kPlatformBrightnessDark;
+    }
```
Comment (cbracken) [inline shell/platform/windows/system_utils_win32.cc:123]: nit: this is fine, but could be slightly less verbose:
```cc
return use_light_theme ? kPlatformBrightnessLight : kPlatformBrightnessDark;
```

File: shell/platform/windows/system_utils_win32.cc [hunk 2]
```diff
@@ -106,4 +106,26 @@ bool Prefer24HourTime(std::wstring time_format) {
   return time_format.find(L"H") != std::wstring::npos;
 }
 
+std::string GetPreferredBrightness() {
```
Comment (cbracken) [inline shell/platform/windows/system_utils_win32.cc:109]: For consistency with the rest of our API (other than `LanguageInfo`), we should use `std::wstring` here.

**Review by cbracken — COMMENTED** [review]
Just one API nit, other than that, this looks good to me!

**Review by cbracken — APPROVED** [review]
LGTM stamp from a Japanese personal seal

**Comment by cbracken** [comment]
Applied a quick one-line format fix. Looks great -- thanks! A lot of people will be excited to see this land! Apologies for the delay in getting to this; was out of the office for a bit.

**Comment by moko256** [comment]
Thanks for your reviewing. (and I'm sorry I forgot to format code.) I'm excited too!

**Comment by cbracken** [comment]
Rebased and pushed to pick up the same API key change as last time.

**Pull request merged**